src/Server/OutputBuilder.cs

```diff
@@ -111,100 +132,64 @@ public OutputBuilder Append(string value)
 
             return this;
         }
-        
+
+        ///<summary>Append bool without memory allocation</summary>
+        public OutputBuilder Append(bool value)
+        {
+            return Append(value ? CharBool[1] : CharBool[0]);
+        }
+
         ///<summary>Append an int without memory allocation</summary>
-        public OutputBuilder Append( int value )
+        public OutputBuilder Append(int value)
         {
-            // Allocate enough memory to handle any int number
-            ReallocateIfn( 16 );
+            return Append((long) value);
+        }
+        
+        ///<summary>Converts int to long to handle int.MinValue</summary>
+        private OutputBuilder Append(long value)
+        {
+            var isNegative = false;
 
-            // Handle the negative case
-            if( value < 0 )
+            if (value < 0)
             {
                 value = -value;
+                isNegative = true;
+            }
+            
+            var length = GetIntLength(value);
+            
+            // Allocate enough memory to handle this int
+            ReallocateIfn(length);
+
+            // Handle the negative case
+            if(isNegative)
                 buffer[ bufferPos++ ] = '-';
+
+            if (value <= 9)
```

Comment:
Noticed this case is superfluous;  0-9 can be easily rolled into the main loop instead of a special case.